### ISSUE TITLE
File stops rendering!

### ISSUE THREAD

**Issue by HaasNL2** [FreeCAD/FreeCAD-render#136]
Finally found this space :) I'm MisterMaker on the forum.
So I been testing cycle render settings.
And so I did a lot of rendering.
And suddenly it stopped working, I got an error.
So I checked if I got the correct xml file by dragging it on the cycles.exe And that works so nothing wrong with my xml files.
But I think what goes wrong is the temp folder where these xml files are stored is running out of characters:
![afbeelding](https://user-images.githubusercontent.com/29804962/112552521-432b9180-8dc3-11eb-9f39-83c8cfcbc3cf.png)
So I think the length of that temp dir should be a bit shorter, although I have no idea what I am doing so could be something completely different issue.

Like lots of testing has been done:
        ```
aa_samples="2"
	ao_samples="100"
	mesh_light_samples="2"
	subsurface_samples="0"
	volume_samples="0"
	diffuse_samples="1"
	glossy_samples="1"
	transmission_samples="0"
	min_bounce="0"
 	max_bounce="8"
	max_diffuse_bounce="4"
	max_glossy_bounce="4"
	max_volume_bounce="0"
	max_transmission_bounce="0"
	sample_clamp_direct="1"
	sample_clamp_indirect="10"
	method="branched_path" 
	sample_all_lights_direct="true" 
	sample_all_lights_indirect="true"
	caustics_reflective="true"
	caustics_refractiv="true"
	seed="98"
	transparent_min_bounce="2"
	transparent_max_bounce="7"
	filter_glossy="1"
	sampling_pattern="sobol"
	/>
```

**Comment by howetuft** [comment]
Hello,
Could you please write the error message?

**Comment by HaasNL2** [comment]
I don't really get an error message it either just crashes immediately or a black screen.
Drmacro also had same issue. He is on Linux so it should be re-producible, just press render couple of times.

I do get a message when I start Freecad, this is my log file:
```
08:42:05  <Exception> PropertyFileIncluded::RestoreDocFile(): File '' in transient directory cannot be created.
08:42:05  <Base> Reader.cpp(747): Reading failed from embedded file: CyclesProjectp42vfbe8.xml
08:42:24  H:/freecad/cyckes/build/bin/Debug/cycles.exe --samples 30 --output C:/Users/Marijn/AppData/Local/Temp/FreeCAD_Doc_cba0a652-50de-4b74-a5db-4427d0842a5e_d672b1_16196/CyclesProjectg3qbex7l.png --width 1920 --height 1080 C:/Users/Marijn/AppData/Local/Temp/FreeCAD_Doc_cba0a652-50de-4b74-a5db-4427d0842a5e_d672b1_16196/CyclesProjectg3qbex7l.xml
```

**Comment by howetuft** [comment]
Maybe a lead: in your second Cycles project, you use the sun-sky template (which contains a sun-sun sky) and a sun-sky light. Apparently, the conjunction of these 2 sun-sky lights makes Cycles crash. Note that it could be Render's fault, I'm not blaming Cycles at this stage. I'll try to have a look at it.
However, at the moment, you should choose: either you use the sun in the template (and you don't add your light to the scene) or you use the light (and you use flat template)...
Does it help?

**Comment by HaasNL2** [comment]
Yes definitely.
I hope that we can use HDRI lighting, then we can drop all lighting options pretty much. Then you can just choose a scene that you want. So if you want to render outside in the grass you van choose grassfield. And studio etc.
I do this also mostly in Blender, only sometimes I still use lights if something is just not right with the current hdri.
You can see HDRI examples here:
https://hdrihaven.com/
No idea if cycles standalone supports this. But if not we can always ask :)

**Comment by howetuft** [comment]
You can use HDRI. It is the 'Image Light' feature, in the light menu. Cycles supports it, so do the other renderers. Enjoy...

**Comment by HaasNL2** [comment]
Ok that is awesome!
So my idea is to make templates with certain HDRI maps. 
Only con at this moment is the path of the template dir is to a user dir, but I thought you already planned to fix this?

**Comment by howetuft** [comment]
Done: I switched Template into relative path.
It is primarily available by default for new projects, as I did not want to break ascending compatibility.
But you can also convert legacy projects to relative path, if you want. It just requires to change the template: the new template will be set in relative path. Nota: changing template has to be done via Project context menu, now.
This way, FreeCAD models with Render projects should be portable from one computer to another.

By the way, I also added a file filter feature: only the templates suitable for the renderer are displayed, instead of all the templates in directory (which was error-prone).

Don't hesitate to give me your feedback!

**Comment by HaasNL2** [comment]
Ok when I pressed the template  browse button it crashed.
Then I removed the sun from the render project and it all worked again.
I think it is smart to remove the sun sky template for now as it crashed freecad.

**Comment by howetuft** [comment]
> I think it is smart to remove the sun sky template for now as it crashed freecad.

Well, no, I won't do that. The crash is not due to the template itself, but to the conjunction of the template and the sun-sky light: there should be a solution to handle this case smartly, I'll have a look next week-end.

**Comment by HaasNL2** [comment]
hmm I had some time to do more testing, also I switched to Linux and then back to Windows (I am not ready for Linux!!). But I did a fresh windows install.
I reinstalled Freecad and downloaded the latest render workbench from here. Not sure if that is necessary but anyway.
 
Now it like always crashes when I press the browse button, this also happens when I press the browse image button when I use image lighting. I check the rights of the folder and everything seems to be right also I am logged in as administrator. 
First it just crashed but after a couple of crashes this warning popped up:
![afbeelding](https://user-images.githubusercontent.com/29804962/113622030-b4d5cc00-965c-11eb-9a00-5ea9e04a67b8.png)
